plugin.video.liveleak/default.py

```diff
@@ -1,196 +1,311 @@
+    liz.setArt( {'thumb': thumbnail} )
+    liz.setProperty('IsPlayable', 'true')
+    cmd = "XBMC.RunPlugin({})"
+    cmd = cmd.format( buildUrl( {'mode': 'mod_user', 'user': credit.encode('utf-8')} ) )
+    liz.addContextMenuItems([('Moderate user: %s' % credit, cmd)])
+
+    return (url, liz)
+
+def saveLeakPosters(leakPosters):
+    # Kludge to force auto-creation of the required userdata folder
+    addon.setSetting('user_prefs', 'enabled')
+    try:
+        with open(leakPostersFileLocation, 'w') as f:
+            f.write(json.dumps(leakPosters))
+        return True
+    except Exception as e:
+        log(e)
+        return False
 
-    return(text)
+def loadLeakPosters():
+    try:
+        with open(leakPostersFileLocation, 'r') as f:
+            return json.loads(f.read())
+    except:
+        saveLeakPosters({})
+        return {}
 
-def addSearch():
-    searchStr = ''
+def getSearchString():
     keyboard = xbmc.Keyboard(searchStr, 'Search')
     keyboard.doModal()
     if (keyboard.isConfirmed()==False):
-        return
-    searchStr=keyboard.getText()
-    if len(searchStr) == 0:
-        return
-    else:
-        return searchStr
-
-def addDir(name, queryString):
-    if 'browse?' not in queryString:
-        queryString = 'browse?' + queryString
-    u="%s?mode=indx&url=%s" % (BASE_URL, qp(queryString))
-    liz=xbmcgui.ListItem(name)
-    liz.setInfo(type="Video", infoLabels={"Title": name})
-    xbmcplugin.addDirectoryItem(handle=ADDON_HANDLE,url=u,listitem=liz,isFolder=True)
-
-def findAllMediaItems(url):
-    page = httpRequest(url)
-    if page is None:
-        notify("The server is not cooperating")
-        return False
-        
-    # Consolidate liveleak and Youtube video sources
-    liveleakRegexp = r'<source src="(.+?)".*$'
-    youtubeRegexp = r'src="//www.youtube.com/embed/(.+?)\?rel=0.*$'
-    Regexp = r'%s|%s' % (liveleakRegexp, youtubeRegexp)
+        return ''
+    return keyboard.getText()
 
-    return re.findall(Regexp, page, re.MULTILINE)
+def addDir(title, qKey, qVal, pVal='1'):
+    url = 'browse?a=list&' + qKey + '=' + qVal + '&page=' + pVal
+    url = buildUrl({'mode': 'indx', 'url': url})
+    liz = xbmcgui.ListItem(title)
+    liz.setInfo(type="Video", infoLabels={"Title": title})
+    xbmcplugin.addDirectoryItem(handle=ADDON_HANDLE,url=url,listitem=liz,isFolder=True)
 
 
 # --- GUI director (Main Event) functions ---
 
 def categories():
-    addDir('Popular', 'popular')
-    addDir('Featured', 'featured=1')
-    addDir('News & Politics', 'channel_token=04c_1302956196')
-    addDir('Yoursay', 'channel_token=1b3_1302956579')
-    addDir('Must See', 'channel_token=9ee_1303244161')
-    addDir('Syria', 'channel_token=cf3_1304149308')
-    addDir('Iraq', 'channel_token=e8a_1302956438')
-    addDir('Afghanistan', 'channel_token=79f_1302956483')
-    addDir('Entertainment', 'channel_token=51a_1302956523')
-    addDir('Search', 'q=')
-    
+    addDir('Featured', 'in_bookmark_folder_id', '1')
```

Comment:
Can you move the hardcoded menu items to a language file?
See: https://kodi.wiki/view/Language_support